### ISSUE TITLE
strange behaviour when restarting trainings

### ISSUE THREAD

**Issue by emilbols** [DL4Jets/DeepJetCore#42]
I have noticed some bizarre effects, seemingly when i restart trainings that crash, or when i try to continue training from a given model.

Here you can see the model trains normally, then at around epoch 22, the training crashed, and i resumed from a checkpoint, and things seemingly goes horribly wrong. The training loss looks normal but validation goes crazy. Any ideas?

[loss_restart.pdf](https://github.com/DL4Jets/DeepJetCore/files/4449703/loss_restart.pdf)

I have only noticed it now in these tf2 version. Do you still have a singularity container for pre tf2 deepjetcore? then i can have a look in that one.

**Comment by jkiesele** [comment]
Thanks for reporting this.
Would it be possible to put together a small script that reproduces the behaviour? 
Even if we end up fixing it, we can add it to the set of tests.

**Comment by emilbols** [comment]
Sure, but i have to first find out in what circumstances it happens (if it happens for all models, datastructures etc.).

**Comment by emilbols** [comment]
This seems to have disappeared in djc3

**Comment by emilbols** [comment]
Actually it seems this is still here:
[StandardReload.pdf](https://github.com/DL4Jets/DeepJetCore/files/4716361/StandardReload.pdf)

Here the training was restarted at around epoch 8 and it seems something goes wrong. However it appears to be connected with the keras.load_model function somehow.

If you instead reload the model by setting the model and then using load_weights, the problems appears to go away..
[LoadWeightsOnly.pdf](https://github.com/DL4Jets/DeepJetCore/files/4716377/LoadWeightsOnly.pdf)

I have been trying to see if this is reproducible using the example dataset that comes with the subpackage. It seems to happen sometimes, but it is kinda hard to tell since the dataset seems a bit too easy to predict.

**Comment by jkiesele** [comment]
It might be related to batch normalisation.
I am following up

**Comment by jkiesele** [comment]
It seems to have disappeared with keras updates.
Could you see if you still see it? (if not we can close)
